internal/cluster/cluster.go

```diff
@@ -826,51 +826,35 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	return nil
 }
 
-func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+func (m *Manager) deleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) error {
 	log := logutil.FromContext(ctx, m.log)
-	files, err := objectHandler.ListObjectsByPrefix(ctx, fmt.Sprintf("%s/logs/", c.ID))
-	if err != nil {
-		return common.NewApiError(http.StatusNotFound, err)
-	}
-
-	var failedToDelete []string
-	for _, file := range files {
-		log.Debugf("Deleting cluster %s S3 log file: %s", c.ID.String(), file)
-		_, err = objectHandler.DeleteObject(ctx, file)
-		if err != nil {
-			m.log.WithError(err).Errorf("failed deleting s3 log %s", file)
-			failedToDelete = append(failedToDelete, file)
-		}
-	}
-
-	if len(failedToDelete) > 0 {
-		return common.NewApiError(
-			http.StatusInternalServerError,
-			errors.Errorf("failed to delete s3 logs: %q", failedToDelete))
+	var path string
+	if folder == "" {
+		path = fmt.Sprintf("%s/", c.ID)
+	} else {
+		path = fmt.Sprintf("%s/%s/", c.ID, folder)
```

Comment:
This could all just be `filepath.Join(c.ID, folder)`. Or `filepath.Join(c.ID, folder) + "/"` if you need the trailing slash.

Reply:
fixed.